src/libcmd/installables.cc

```diff
@@ -617,17 +617,49 @@ DerivedPathsWithInfo InstallableFlake::toDerivedPaths()
         }
 
         else if (v.type() == nString) {
-            PathSet context;
-            auto s = state->forceString(v, context, noPos, fmt("while evaluating the flake output attribute '%s'", attrPath));
-            auto storePath = state->store->maybeParseStorePath(s);
-            if (storePath && context.count(std::string(s))) {
-                return {{
-                    .path = DerivedPath::Opaque {
-                        .path = std::move(*storePath),
-                    }
-                }};
-            } else
-                throw Error("flake output attribute '%s' evaluates to the string '%s' which is not a store path", attrPath, s);
+            auto context = v.getContext(*state->store);
+            std::string_view s = state->forceString(v, noPos, fmt("while evaluating the flake output attribute '%s'", attrPath));
+            if (context.size() != 1)
+                throw Error(
+                    "flake output attribute '%s' evaluates to the string '%s' which does not denote exactly one store",
+                    attrPath, s);
+            return {{
+                .path = std::visit(overloaded {
+                    [&](NixStringContextElem::Opaque & o) -> DerivedPath {
+                       auto sExpected = state->store->printStorePath(o.path);
+                       if (s != sExpected)
+                           throw Error(
+                               "flake output attribute '%s' evaluates to the string '%s' which should be '%s' but is not",
+                               attrPath, s, sExpected);
+                       return DerivedPath::Opaque {
+                           .path = std::move(o.path),
+                       };
+                    },
+                    [&](NixStringContextElem::DrvDeep & o) -> DerivedPath {
+                       throw Error(
+                           "flake output attribute '%s' evaluates to the string '%s' which unsupported String Context",
+                           attrPath, s);
+                    },
+                    [&](NixStringContextElem::Built & o) -> DerivedPath {
+                       auto drv = state->store->readDerivation(o.drvPath);
+                       auto i = drv.outputs.find(o.output);
+                       if (i == drv.outputs.end())
+                           throw Error("derivation '%s' does not have output '%s'", state->store->printStorePath(o.drvPath), o.output);
+                       auto optOutputPath = i->second.path(*state->store, drv.name, o.output);
+                       auto sExpected = optOutputPath
```

Comment:
```suggestion
                       # This is testing for the case of CA derivations
                       auto sExpected = optOutputPath
```
and we should confirm that this derivation is indeed CA.

Reply:
```suggestion
                       // This is testing for the case of CA derivations
                       auto sExpected = optOutputPath
```